pytorch_lightning/core/lightning.py

```diff
@@ -1976,4 +1977,6 @@ def _register_sharded_tensor_state_dict_hooks_if_available(self) -> None:
         from torch.distributed._sharded_tensor import pre_load_state_dict_hook, state_dict_hook
 
         self._register_state_dict_hook(state_dict_hook)
-        self._register_load_state_dict_pre_hook(pre_load_state_dict_hook, True)
+
+        # We need to make sure the self inside the method is a weakref proxy
+        Module._register_load_state_dict_pre_hook(weakref.proxy(self), pre_load_state_dict_hook, True)
```

Comment:
Nice hack!

Might be overkill, but I would suggest getting the parent class dynamically

```suggestion
        parent = inspect.getmro(self.__class__)[1]
        parent._register_load_state_dict_pre_hook(weakref.proxy(self), pre_load_state_dict_hook, True)
```

Reply:
Is it possible that the parent would be anything other than Module? And do we even want to invoke it if it is something else than Module? I can imagine in user defined classes which inherits from LightningModule, this would actually resolve to LightningModule, right?